h/form.py

```diff
@@ -15,16 +20,26 @@
 )
 
 
-def init():
-    from deform import Form
-    from deform_jinja2 import jinja2_renderer_factory
-    from deform_jinja2.translator import PyramidTranslator
+# We supply our own Jinja2 renderer factory because deform_jinja2's one doesn't
+# support autoescaping.
+def jinja2_renderer_factory(config):
+
+    def jinja2_renderer(template_name, **kwargs):
+        searchpath = [pkg_resources.resource_filename(*(path.split(':')))
+                      for path in SEARCH_PATHS]
+
+        environment = config.get_jinja2_environment().overlay(
```

Comment:
You're still creating a new overlay environment for every single template render call.

On approach to fixing this would be to move this up into the body of the `jinja2_renderer_factory` function and close over it.